### PR TITLE
Collapsible Component Tests

### PR DESCRIPTION
**Pull request by galloween**
all, except Event outputs

### CODE REVIEW

File: projects/ui-framework/src/lib/layout/collapsible/collapsible.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      .then(() => {
+        fixture = TestBed.createComponent(TestComponent);
+        component = fixture.componentInstance;
+        spyOn(component.opened, 'emit');
+        spyOn(component.closed, 'emit');
+      });
+  }));
+
+  describe('component', () => {
+    it(': should start with no content (content lazy loaded on first panel open)', () => {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/layout/collapsible/collapsible.component.spec.ts]: remove :

File: projects/ui-framework/src/lib/layout/collapsible/collapsible.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      fixture.detectChanges();
+      const panelBodyElement = fixture.debugElement.query(
+        By.css('.mat-expansion-panel-body')
+      );
+      expect(panelBodyElement.nativeElement.children.length).toEqual(0);
+    });
+
+    it(': if the panel is expanded, we should see content (this tests both the expanded input and the lazy content init mechanism)', () => {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/layout/collapsible/collapsible.component.spec.ts]: should start it with should

File: projects/ui-framework/src/lib/layout/collapsible/collapsible.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+    it(': element with attribute [suffix] should be transcluded in the header', () => {
+      fixture.detectChanges();
+      const suffixElement = fixture.debugElement.query(
+        By.css('.mat-expansion-panel-header .collapsible-suffix [suffix]')
+      );
+      expect(suffixElement).toBeTruthy();
+    });
+
+    it(': type=small input should display title input in b-subheading element', () => {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/layout/collapsible/collapsible.component.spec.ts]: group content in describe.
```
describe('title', () => {
it(....
});
```

File: projects/ui-framework/src/lib/layout/collapsible/collapsible.component.ts [hunk 4]
```diff
@@ -39,8 +39,13 @@ export class CollapsibleComponent implements AfterViewInit {
   hasSuffix = true;
 
   ngAfterViewInit() {
-    this.hasSuffix =
-      this.suffix.nativeElement.children.length !== 0 ? true : false;
+    // wait a tick first to avoid one-time devMode
+    // unidirectional-data-flow-violation error
+    // https://stackoverflow.com/a/38937802
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/layout/collapsible/collapsible.component.ts]: remove comments

File: projects/ui-framework/src/lib/layout/collapsible/collapsible.stories.ts [hunk 5]
```diff
@@ -62,11 +62,24 @@ const note = `
   disabled | boolean | if the panel is disabled (can't be opened) | false
   title | panel header title | title | ''
   description | string | panel header description | none (optional)
-  opened | Function | Event emitted every time the panel is opened
-  closed | Function | Event emitted every time the panel is closed
 
   Content marked with [suffix] will be projected into the right part of the panel header.
 
+
+  #### Events
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/layout/collapsible/collapsible.stories.ts]: if this is not exposed we don't need it in the storybook
exposed = Input() & Output()

File: projects/ui-framework/src/lib/layout/collapsible/collapsible.stories.ts [hunk 6]
```diff
@@ -62,11 +62,24 @@ const note = `
...
+  opened | Emitted every time the panel is opened
+  closed | Emitted every time the panel is closed
+
+
+  #### Methods
+  Name | Description
+  --- | ---
+  close() | collapses panel
+  open() | expands panel, if collapsible is not disabled
+  toggle() | expands or collapses panel, if collapsible is not disabled
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/layout/collapsible/collapsible.stories.ts]: toggle is not exposed, need to remove it from the stories

File: projects/ui-framework/src/lib/layout/collapsible/collapsible.component.ts [hunk 7]
```diff
@@ -38,16 +38,34 @@ export class CollapsibleComponent implements AfterViewInit {
...
+      this.expanded = true;
+    }
   }
 
-  onPanelOpened($event) {
+  close(): void {
+    this.expanded = false;
+  }
+
+  toggle(): void {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/layout/collapsible/collapsible.component.ts]: who calls this method?

Comment (galloween) [reply projects/ui-framework/src/lib/layout/collapsible/collapsible.component.ts]: its public. I just duplicated API from Material. open, close and toggle methods can be used by consumer, if needed. but of course, they are just wrappers around the [expanded] prop

**Pull request merged**